packages/wallet/src/redux/protocols/withdrawing/readme.md

```diff
@@ -0,0 +1,34 @@
+# Withdrawing Protocol
+
+The purpose of this protocol is to withdraw funds from a directly funded channel.
+It covers:
+
+- Checking that a channel is closed.
+- Displaying to the user the funds allocated to them in the channel,
+- Getting approval from the user to proceed
+- Sending the withdraw + conclude to the blockchain
+- Getting acknowledgement from the user when done
+
+Out of scope (for the time being):
+
+- Retrying a transaction on failure.
+- Displaying an error message to the user on transaction failure.
+- Checking if a finalized outcome exists on chain(This will probably require a new saga).
+
+## State machine
+
+The protocol is implemented with the following state machine
+
+```mermaid
+graph LR
+  St((start)) --> WFAp(WaitForApproval)
+  WFAp --> |Approved| WFT(WaitForTransaction)
+  WFT --> |TransactionSubmitted| WFAk(WaitForAcknowledgement)
+  WFT --> |TransactionFailed| F((failure))
+  WFAk --> |SuccessAcknowledged| S((success))
+  WFAp --> |Rejected| F((failure))
+```
```

Comment:
I think you need at least one more state here to handle the situation where the user tries to withdraw before the channel is closed (e.g. either a conclusion proof exists or the channel has been finalized on-chain). It probably makes sense to dump the user straight into that state instead of `WaitForApproval`. It would probably be `AcknowledgeNotClosed` or something.

Reply:
I think that makes sense. Although practically I don't think we should ever let the user get there. We should only start the withdrawing protocol when we know the channel is closed.

Reply:
That's a good point. It kind depends how this protocol is going to be triggered, which isn't completely clear yet. Happy if you want to leave it out for now.

Reply:
I think for now I'll put the channel closed check in initialize and dump into the failure state if its not. If we want to add a new acknowledge state that will be trivial to do in the future.